packages/glimmer-runtime/lib/syntax/builtins/partial.ts

```diff
@@ -0,0 +1,79 @@
+import { assert } from "glimmer-util";
+
+import {
+  CompileInto,
+  SymbolLookup,
+  Statement as StatementSyntax
+} from '../../syntax';
+
+import Scanner from '../../scanner';
+
+import {
+  LabelOpcode,
+  EnterOpcode,
+  PutArgsOpcode,
+  TestOpcode,
+  JumpUnlessOpcode,
+  JumpOpcode,
+  EvaluatePartialOpcode,
+  ExitOpcode
+} from '../../compiled/opcodes/vm';
+
+import * as Syntax from '../core';
+import Environment from '../../environment';
+
+export default class PartialSyntax extends StatementSyntax {
+  type = "partial-statement";
+
+  public args: Syntax.Args;
+  public isStatic = false;
+
+  constructor({ args }: { args: Syntax.Args }) {
+    super();
+    this.args = args;
+  }
+
+  prettyPrint() {
+    return `partial ${this.args.prettyPrint()}`;
+  }
+
+  compile(compiler: CompileInto & SymbolLookup, env: Environment) {
+
+    /*
+    //        Enter(BEGIN, END)
+    // BEGIN: Noop
+    //        PutArgs
+    //        Test
+    //        JumpUnless(ELSE)
+    //        Evaluate(default)
+    //        Jump(END)
+    // ELSE:  Noop
+    //        Evaluate(inverse)
+    // END:   Noop
+    //        Exit
+    */
+
+    assert(this.args.positional.values.length > 0, `Partial found with no arguments. You must specify a template.`);
+    assert(this.args.positional.values.length < 2, `Partial found with more than one argument. You can only specify a single template.`);
+
+    let compiledPartialNameExpression = this.args.positional.values[0].compile(compiler, env);
+
+    let BEGIN = new LabelOpcode({ label: "BEGIN" });
+    let ELSE = new LabelOpcode({ label: "ELSE" });
+    let END = new LabelOpcode({ label: "END" });
+
+    compiler.append(new EnterOpcode({ begin: BEGIN, end: END }));
+    compiler.append(BEGIN);
+    compiler.append(new PutArgsOpcode({ args: this.args.compile(compiler, env) }));
+    compiler.append(new TestOpcode());
```

Comment:
There is a missing step here. I assume this is trying to test that if the partial name is falsy, don't render anything. However, the env can fail to lookup the partial for other reasons. So I think `PutArgs` is fine, but you should insert another opcode here that converts the name partial into a `PartialDefinition` before running the `TestOpcode`?